### ISSUE TITLE
properties clear on GM restart

### ISSUE THREAD

**Issue by Y-Less** [Open-GTO/sa-mp-fixes#98]
They weren't designed for systems that are long-running.

**Comment by Daniel-Cortez** [comment]
I think a proper explanation is that properties in Pawn weren't designed for running more than one script at a time.

What happens is that when a script (and by that I mean literally any script, not only a GM as stated in the title) gets unloaded, `amx_CoreCleanup()` removes _all_ properties, even those that were created by the other scripts.

https://github.com/pawn-lang/compiler/blob/master/source/amx/amxcore.c#L509-L512
In Pawn 3.0 there also was a comment in that part of code
```C
  #if !defined AMX_NOPROPLIST
    //??? delete only the properties owned by the AMX
    while (proproot.next!=NULL)
      list_delete(&proproot,proproot.next);
  #endif
```
which means the original Pawn creator probably intended to delete only the properties created by the currently unloaded script, but judging by removal of this comment in 3.1/3.2 the idea was scrapped for some reason.

**Comment by Y-Less** [comment]
Yeah, I forgot that filterscripts could also be (un)loaded at runtime.
